applications/HDF5Application/python_scripts/core/file_io.py

```diff
@@ -134,11 +134,11 @@ def Get(self, model_part=None):
         return file_name
 
     def _InitializeDirectory(self, file_name):
-        dirname = os.path.dirname(file_name)
-        if dirname != '' and not os.path.exists(dirname):
-            if self.data_comm.Rank() == 0:
-                os.makedirs(dirname)
-            self.data_comm.Barrier()
+        dirname = Path(file_name).absolute().parent
+        if self.data_comm.IsDistributed():
+            KratosMultiphysics.FilesystemExtensions.MPISafeCreateDirectories(str(dirname))
+        else:
+            dirname.mkdir(exist_ok=True, parents=True)
```

Comment:
```suggestion
        KratosMultiphysics.FilesystemExtensions.MPISafeCreateDirectories(str(dirname))
```

it can be safely used either way

Reply:
Updated. Weren't there concerns about entering infinite loops on access-related errors?

Reply:
those should hopefully be resolved
In any case if more users test it we might be able to fix it sooner in case it is a problem
I much rather prefer sth to fail locally and not on the cluster